### PR TITLE
improvement(test_latency_with_nemesis): display events to email report

### PR DESCRIPTION
**Pull request by fgelcer**
improvement(test_latency_with_nemesis): display events to email report

to improve understanding the email report, adding now
the events (like we have in the longevity tests) to
be able to identify easier failures or potential issues
in the test from the email report.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fgelcer** [comment]
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fabio/job/latency_automatic_calculation_read/

**Comment by fgelcer** [comment]
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fabio/job/latency_automatic_calculation_read/16/

File: sdcm/report_templates/results_latency_during_ops.html [hunk 1]
```diff
@@ -61,6 +61,23 @@ <h2>{{ operation }}</h2>
...
+                </h4>
+                {% for event in events %}
+                    <pre>{{event}}</pre>
+                    <hr>
+                {% endfor %}
+            {% endfor %}
+        {% endif %}
+    </div>
+
     <h3>Links:</h3>
```
Comment (aleksbykov) [inline sdcm/report_templates/results_latency_during_ops.html:81]: if this code repeated in several  templates, may be  it is better to move it to separate file and just include it here?

File: sdcm/results_analyze/__init__.py [hunk 2]
```diff
@@ -157,8 +157,21 @@ def check_regression(self, test_id, data, is_gce=False):  # pylint: disable=too-
...
+        events_summary = dict()
+        last_events['CRITICAL'] = doc["_source"]["events"]["CRITICAL"]
+        events_summary['CRITICAL'] = len(last_events['CRITICAL'])
+
+        last_events['ERROR'] = doc["_source"]["events"]["ERROR"]
+        events_summary['ERROR'] = len(last_events['ERROR'])
+
+        last_events['WARNING'] = doc["_source"]["events"]["WARNING"]
+        events_summary['WARNING'] = len(last_events['WARNING'])
+        subject = f'Performance Regression Compare Results (latency during mgmt ops) -' \
```
Comment (aleksbykov) [inline sdcm/results_analyze/__init__.py:170]: I think, new method which will return ready dicts will be better. Also you can add some checks there if such fields by some reasons will be missed in document.

**Review by aleksbykov — COMMENTED** [review]
LGTM, with small comments

**Comment by aleksbykov** [comment]
@fgelcer , please address comments

**Comment by fgelcer** [comment]
> @fgelcer , please address comments

i'm on it

File: sdcm/report_templates/results_latency_during_ops.html (hunk 1 above)
Comment (fgelcer) [reply sdcm/report_templates/results_latency_during_ops.html:81]: this was not yet done (i actually started, but didn't finish, and didn't test other templates that "will use it"

File: sdcm/results_analyze/__init__.py (hunk 2 above)
Comment (fgelcer) [reply sdcm/results_analyze/__init__.py:170]: this was done, and is being tested [here](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fabio/job/latency_automatic_calculation_read/17/)

File: sdcm/report_templates/results_latency_during_ops.html (hunk 1 above)
Comment (fgelcer) [reply sdcm/report_templates/results_latency_during_ops.html]: change was done, i'm checking for regression on other templates that use the `results_base_custom.html` too

File: sdcm/results_analyze/__init__.py [hunk 3]
```diff
@@ -117,48 +167,26 @@ def __init__(self, es_index, es_doc_type, send_email, email_recipients, logger=N
...
-        if not grafana_screenshots:
-            grafana_screenshots = test_doc['_source']['test_details'].get('grafana_screenshots')
-
-        if grafana_screenshots and isinstance(grafana_screenshots, list):
-            return grafana_screenshots
-        elif grafana_screenshots and isinstance(grafana_screenshots, str):
-            return [grafana_screenshots]
-        else:
-            return []
+        return result_details
```
Comment (aleksbykov) [inline sdcm/results_analyze/__init__.py:173]: @fgelcer this method should return empty dict?

**Review by aleksbykov — APPROVED** [review]
LGTM, Small comments

File: sdcm/results_analyze/__init__.py (hunk 3 above)
Comment (fgelcer) [reply sdcm/results_analyze/__init__.py]: ops, it was a leftover from previous code, and it has moved to other functions, hence it was now removed.

**Review by aleksbykov — APPROVED** [review]
LGTM

**Comment by fgelcer** [comment]
@fruch @roydahan , ready to merge

**Comment by fgelcer** [comment]
@roydahan , we need this in the branch-perf-v10, otherwise, it won't be in use for the weekend run of latency_with_nemesis tests

**Comment by roydahan** [comment]
Backported, please run a sample job to make sure it works.

**Comment by fgelcer** [comment]
> Backported, please run a sample job to make sure it works.

running https://jenkins.scylladb.com/job/scylla-master/job/scylla-master-perf-regression-latency-250gb-with-nemesis/28/

**Pull request merged**